### ISSUE TITLE
Doesn't work in LibreELEC v7.90.005 Alpha and the latest OpenELEC

### ISSUE THREAD

**Issue by LaurensVanAcker** [kodi-pvr/pvr.dvblink#53]
Hi

Enabling the DVBLogic plugin in LibreElec brings Kodi in an endless reboot loop. I enabled version 3.3.4. It used to work fine a couple of weeks ago in OpenElec, but it's doing this now for about 2 weeks. It did this in OpenELEC 7.0 Beta 3 and I just did a full clean reinstall with v7.90.005 Alpha where it does just the same as it did in OpenELEC.

Attached some debug logs. For example I do see messages like this:

Sep 03 16:54:29 Kodi kernel: PVRManager[881]: segfault at 20 ip 00007f2ea808cfca sp 00007f2e5efaf920 error 4 in pvr.dvblink.so.3.3.4[7f2ea804c000+68000]

Program terminated with signal SIGSEGV, Segmentation fault.
#0 0x00007f2ea808cfca in DVBLinkClient::GetSchedules(ADDON_HANDLE_STRUCT*, dvblinkremote::RecordingList const&) () from /storage/.kodi/addons/pvr.dvblink/pvr.dvblink.so.3.3.4

[Current thread is 1 (Thread 0x7f2e5efb2700 (LWP 881))]

[Archief.zip](https://github.com/kodi-pvr/pvr.dvblink/files/453460/Archief.zip)
I'm not running it on a Raspberry Pi, just on generic hardware.

Greetings
Laurens

**Comment by LaurensVanAcker** [comment]
This is with the #0902 build (02-Sep-2016)
[log-2016-09-03-18.12.25.zip](https://github.com/kodi-pvr/pvr.dvblink/files/453529/log-2016-09-03-18.12.25.zip)

**Comment by LaurensVanAcker** [comment]
[crashlog.txt](https://github.com/kodi-pvr/pvr.dvblink/files/453532/crashlog.txt)

**Comment by dvblogic** [comment]
Did you have any "keyword search" recording timers? If yes, the issue should be fixed by a pending pull request "pvr.dvblink version 3.3.5 (krypton) #54"

**Comment by Jalle19** [comment]
Most likely fixed by https://github.com/kodi-pvr/pvr.dvblink/pull/56

**Comment by Jalle19** [comment]
Actually that fix was only for Jarvis, reopening.

**Comment by Jalle19** [comment]
@LaurensVanAcker is this still an issue with the latest LibreELEC builds (assuming they ship with the pvr.dvblink version 3.3.5)?

**Comment by LaurensVanAcker** [comment]
I tried installing 3.3.5, but Kodi crashes and reboots. Now 3.3.4 also stopped working with errors like this:

08:24:06  33.888912 T:140143590717184   ERROR: CommitInsertQueries - failed to execute queries
08:24:06  33.888920 T:140143590717184   ERROR: CPVRChannelGroups - Load - failed to load channel groups
08:24:06  33.888924 T:140143590717184   ERROR: PVRManager - Process - failed to load PVR data, retrying
08:24:07  34.916481 T:140143590717184   ERROR: SQL: [TV29.db] Abort due to constraint violation
                                            Query: INSERT INTO channels (iUniqueId, bIsRadio, bIsHidden, bIsUserSetIcon, bIsUserSetName, bIsLocked, sIconPath, sChannelName, bIsVirtual, bEPGEnabled, sEPGScraper, iLastWatched, iClientId, idEpg) VALUES (28, 0, 0, 0, 0, 0, 'http://nova.nempo.com:8101/dvblink/playback?object=c4bb498b-0542-46d8-a13d-59abef042af40x0.000000116023p-1022Andere0.000000nat-geo-wild.png', 'Nat Geo Wild HD NL', 0, 1, 'client', 0, 1134691448, -1)

**Comment by LaurensVanAcker** [comment]
Debug log of the crash: http://pastebin.com/3hKGVnYJ

**Comment by Jalle19** [comment]
Yeah that's a different issue we're trying to fix.

**Comment by Jalle19** [comment]
Actually the crash is not what I thought it was, but the errors from your snippet are relates to this other issue.

**Comment by LaurensVanAcker** [comment]
I upgraded to libreELEC v7.90.006 and cleared the pvr.dvblink folder under user date.
Now I have version 3.3.5. Unfortunately I'm still blocked on this error:

```
21:21:51 T:139758569211648   DEBUG: PVRChannelGroup - Load - 3 channels added from clients to group 'All channels'
21:21:51 T:139758569211648   DEBUG: CPVRChannelGroups - LoadUserDefinedChannelGroups - 0 new user defined TV channel groups fetched from clients
21:21:51 T:139758569211648   DEBUG: CPVRChannelGroups - PersistAll - persisting all changes in channel groups
21:21:51 T:139758569211648   DEBUG: CPVRChannelGroup - Persist - persisting channel group 'All channels' with 36 channels
21:21:51 T:139758569211648   ERROR: SQL: [TV29.db] Abort due to constraint violation
                                            Query: INSERT INTO channels (iUniqueId, bIsRadio, bIsHidden, bIsUserSetIcon, bIsUserSetName, bIsLocked, sIconPath, sChannelName, bIsVirtual, bEPGEnabled, sEPGScraper, iLastWatched, iClientId, idEpg) VALUES (28, 0, 0, 0, 0, 0, 'http://nova.nempo.com:8101/dvblink/playback?object=c4bb498b-0542-46d8-a13d-59abef042af40x0.0000001165494p-1022Andere0.000000nat-geo-wild.png', 'Nat Geo Wild HD NL', 0, 1, 'client', 0, 1134691448, -1)
21:21:51 T:139758569211648   ERROR: CommitInsertQueries - failed to execute queries
21:21:51 T:139758569211648   ERROR: CPVRChannelGroups - Load - failed to load channel groups
21:21:51 T:139758569211648   ERROR: PVRManager - Process - failed to load PVR data, retrying
```

**Comment by LaurensVanAcker** [comment]
Got it working by removing the tv.db in the userdata. Also I removed all Search Patterns in DVBLogic with "-signs in it. That's a problem that they are fixing for the next server release. It's only still buffering the IP streams a lot (and not on other computers), but that's a separate issue.
